Tidy up some (rarely touched) tests #4235

wkurniawan07 opened the pull request:
Fixes #4235 
How (a portion of) the test package looks like now:
![test_package](https://cloud.githubusercontent.com/assets/7261051/9669624/f93072ec-52ba-11e5-9ac3-8338d2058a79.png)
And how (again, just a portion of) `testng.xml` looks now:
![test_testng](https://cloud.githubusercontent.com/assets/7261051/9669598/d2684450-52ba-11e5-81c1-49e2607db990.png)
Much neater IMO. Now all test cases are found only in `test.cases.*` package. In fact, all the test files match the regex `test\/cases\/[a-z]+(\/[a-z]+)*\/(?!\QBase\E)[A-Za-z]+\.java`, if anyone wants to test it that way.

yj-soh commented:
Should the callback classes be moved out of their respective classes?
If the classes are exclusively for those they were in, it may be better to leave them there to denote that they aren't meant to be used elsewhere.

yj-soh commented on src/test/java/teammates/test/automated/FeedbackSessionsEmailTaskQueueCallback.java:
```diff
@@ -0,0 +1,31 @@
+import com.google.appengine.api.urlfetch.URLFetchServicePb.URLFetchRequest;
+
+@SuppressWarnings("serial")
+public class FeedbackSessionsEmailTaskQueueCallback extends BaseTaskQueueCallback {
```
Either include or eliminate the `TaskQueue` part in all of these classes?
I see you eliminated it from `SubmissionsAdjustmentTaskQueueCallback`, but it's still present here.

wkurniawan07 commented:
I'm personally not into nested classes, unless they're really needed (like for encapsulation purpose and stuff).
But then again, it's true that they're not used anywhere else. Guess I'll just put them back in and make them private (they were public before).

wkurniawan07 commented:
I reverted the renames. I realized that if I renamed them the way I wanted it to be (the callback classes' names follow the `*` in `*Action` and the test classes' names follow the `*` in `*Servlet`, as per the `logic.automated` package), the PR will be too big. Not that important anyway.
